### PR TITLE
KotlinGenerator: put option fields generation behind an plugin option…

### PR DESCRIPTION
**Pull request by oldergod**
… defaulted to false.

### CODE REVIEW

File: wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt [hunk 1]
```diff
@@ -182,7 +182,10 @@ data class KotlinTarget(
   val rpcRole: RpcRole = RpcRole.CLIENT,
 
   /** True for emitted services to implement one interface per RPC. */
-  val singleMethodServices: Boolean = false
+  val singleMethodServices: Boolean = false,
+
+  /** True for option fields of emitted types' members to be generated. */
+  val generateOptionFields: Boolean = false
```
Comment (Egorand) [inline wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt]: I don't like this in the API. As a consumer of this library, it's not clear to me when and why I should use it. I just want stuffs to work.

Comment (oldergod) [reply wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt]: Would you rather just hide this feature for now? never generate those for Kotlin until we meet someone with a use case ?

Comment (Egorand) [reply wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt]: If the feature is broken we should fix it.

**Comment by oldergod** [comment]
We're looking into getting rid of those static fields so obsolete.

**Pull request closed without merging**